packages/material-ui/src/Menu/Menu.js

```diff
@@ -126,16 +200,70 @@ const Menu = React.forwardRef(function Menu(props, ref) {
       });
     }
 
+    // If the current item in this map has nestedItems,
+    // we need the Menu to orchestrate its nestedMenu
+    if (hasNestedMenu && parentMenuOpen) {
+      additionalPropsAdded = true;
+
+      const handleArrowRightKeydown = (event) => {
+        if (event.key === 'ArrowRight') {
+          event.preventDefault();
+          setLastEnteredItemIndex(index);
+        }
+      };
+      
+      Object.assign(additionalProps, {
+        handleArrowRightKeydown,
```

Comment:
Why do the children need the arrowRightKeyDown but the list needs arrowLeft?

Reply:
Because, when you do an ArrowRight, you're on the parent MenuItem but, when you do an ArrowLeft, you're on the child Menu. Passing the ArrowLeft handler in with the pre-existing handler function that was there for the 'Tab' behavior is just a convenient way to get that event handler onto the subMenu.

If that's not very helpful, let me know and I can probably walk you through it with a demonstration a little easier than I can type an explaination.